app/src/main/java/org/dhis2/usescases/eventsWithoutRegistration/eventCapture/EventCapturePresenterImpl.java

```diff
@@ -606,16 +606,7 @@ public boolean getCompletionPercentageVisibility() {
 
     @Override
     public void setValueChanged(@NotNull String uid) {
-        compositeDisposable.add(
-                Completable.fromCallable(() -> {
-                    eventCaptureRepository.updateFieldValue(uid);
-                    return true;
-                })
-                        .subscribeOn(schedulerProvider.io())
-                        .observeOn(schedulerProvider.io())
```

Comment:
why changing threads?

Reply:
We needed to be a blocking process so that the list is not returned before the value is updated